src/pages/GroupInfo/GroupInfo.js

```diff
@@ -147,6 +152,24 @@ class GroupInfo extends Component {
               }
             />
 
+            {data.archived &&
+              <Row>
+                <Col lg={12}>
+                  <p className="callout callout-warning">
+                    <FormattedMessage
+                      id="app.group.archivedExplain"
+                      defaultMessage="This group is archived, so it cannot be modified."
+                    />
+                    <br />
+                    {!data.directlyArchived &&
+                      <FormattedMessage
+                        id="app.group.notDirectlyArchived"
+                        defaultMessage="This group inherits archivation flag from one of its parent groups."
+                      />}
+                  </p>
+                </Col>
+              </Row>}
```

Comment:
Actually, I think I have seen this fragment at least three times already. The rule of three states that this should have been refactored into a shared component.